internal/services/netapp/netapp_volume_quota_rule_resource.go

```diff
@@ -0,0 +1,314 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package netapp
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/netapp/2022-05-01/volumequotarules"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/netapp/2022-05-01/volumes"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	netAppModels "github.com/hashicorp/terraform-provider-azurerm/internal/services/netapp/models"
+	netAppValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/netapp/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type NetAppVolumeQuotaRuleResource struct{}
+
+var _ sdk.Resource = NetAppVolumeQuotaRuleResource{}
+
+func (r NetAppVolumeQuotaRuleResource) ModelObject() interface{} {
+	return &netAppModels.NetAppVolumeQuotaRuleModel{}
+}
+
+func (r NetAppVolumeQuotaRuleResource) ResourceType() string {
+	return "azurerm_netapp_volume_quota_rule"
+}
+
+func (r NetAppVolumeQuotaRuleResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return volumequotarules.ValidateVolumeQuotaRuleID
+}
+
+func (r NetAppVolumeQuotaRuleResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: netAppValidate.VolumeQuotaRuleName,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"account_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: netAppValidate.AccountName,
+		},
+
+		"pool_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: netAppValidate.PoolName,
+		},
+
+		"volume_name": {
```

Comment:
Instead of the attributes `resource_group_name`, `account_name`, `pool_name`, and `volume_name`, can we instead use `netapp_volume_id`? It'll shorten everything quite a bit

Reply:
Sounds good to me, but I will do volume_id instead since the focus is Azure Netapp Files resource and I don't think we should add netapp as prefix, it could make sense for a service external to mine like AVS but there it will be odd.